### DISCUSSION TITLE
unable to add custom elements to TinyMCE

### DISCUSSION THREAD

**Discussion by atomGit** [GetPublii/Publii#1870] [Help]
followed instructions [here](https://gist.github.com/candidexmedia/367711a80274b4c005fd5983df97d20c#instructions) and changed this...

```
            "selector": "table"
        }
    ]
}
```

to this...
```
            "selector": "table"
        },
        {
            "title": "Superscript1",
            "selector": "sup"
        },
        {
            "title": "Subscript1",
            "selector": "sub"
        }
    ]
}
```

entire TinyMCE toolbar disappears

**Comment by candidexmedia** [comment]
Hi, gist author here! Are you creating Custom Elements, Custom Formats, or Custom Blocks? Which file are you editing? Each requires different changes.

**Comment by atomGit** [comment]
i'm trying to add the superscript and subscript elements exactly as outlined [here](https://gist.github.com/candidexmedia/367711a80274b4c005fd5983df97d20c)

the file is the same as in the instructions: `/Publii/sites/[siteName]/input/themes/simple/config.json`

**Reply by candidexmedia** [reply]
Can you share the file here so I can have a look?

**Comment by atomGit** [comment]
see the bottom of the file, in the `customElements` section

[config.json](https://github.com/user-attachments/files/18112994/config.json)

**Reply by candidexmedia** [reply]
Review the instructions for **CustomElementsMode** in the gist again carefully: https://gist.github.com/candidexmedia/367711a80274b4c005fd5983df97d20c#instructions

You are currently using the method for **[Custom Formats](https://gist.github.com/candidexmedia/367711a80274b4c005fd5983df97d20c#file-3-custom-formats-blocks-md)**, which are for adding CSS classes to certain HTML tags:

![Screenshot](https://github.com/user-attachments/assets/da58cf4c-bd49-466b-8e6b-c0297c82b9b2)

Instead, it should look like this (notice the customElementsMode and different syntax):

![Screenshot_20241212-105446](https://github.com/user-attachments/assets/af57de77-176a-47dd-91e7-96560b250c8f)

Replace your customElements with the example in my gist, restart your app, and check again.

The gist goes over several different ways of modifying the editor, and they each have their own syntax, quirks, nuance, and sets of instructions.

**Comment by atomGit** [comment]
yeah, i see my mistake - thanks for the tip :)
